src/JsonLd/Serializer/ItemNormalizer.php

```diff
@@ -91,20 +92,33 @@ public function supportsDenormalization($data, $type, $format = null)
 
     /**
      * {@inheritdoc}
-     *
-     * @throws InvalidArgumentException
      */
-    public function denormalize($data, $class, $format = null, array $context = [])
+    protected function setObjectToPopulate($data, string $class, array &$context)
     {
-        // Avoid issues with proxies if we populated the object
-        if (isset($data['@id']) && !isset($context[self::OBJECT_TO_POPULATE])) {
-            if (isset($context['api_allow_update']) && true !== $context['api_allow_update']) {
-                throw new InvalidArgumentException('Update is not allowed for this operation.');
+        try {
+            parent::setObjectToPopulate($data, $class, $context);
+        } catch (InvalidArgumentException $e) {
+            // https://github.com/api-platform/core/issues/857
+            $identifiers = $this->identifiersExtractor->getIdentifiersFromResourceClass($class);
+            $identifiersData = \array_intersect_key($data, array_flip($identifiers));
+            if (0 === \count($identifiersData)) {
```

Comment:
```php
if (!$identifiersData) {
```

Reply:
is this a style adopted by API-P? i prefer to be explicit but happy to align with project style.